packages/@react-aria/dnd/src/useDrop.ts

```diff
@@ -158,30 +159,43 @@ export function useDrop(options: DropOptions): DropResult {
     e.preventDefault();
     e.stopPropagation();
 
-    if (typeof options.onDrop === 'function') {
-      let dropOperation = DROP_EFFECT_TO_DROP_OPERATION[state.dropEffect];
-      let items = readFromDataTransfer(e.dataTransfer);
-
-      let rect = (e.currentTarget as HTMLElement).getBoundingClientRect();
-      let event: DropEvent = {
-        type: 'drop',
-        x: e.clientX - rect.x,
-        y: e.clientY - rect.y,
-        items,
-        dropOperation
-      };
-
-      // Wait a frame to dispatch the drop event so that we ensure the dragend event fires first.
-      // Otherwise, if onDrop removes the original dragged element from the DOM, dragend will never be fired.
-      // This behavior is consistent across browsers, but see this issue for details:
-      // https://bugzilla.mozilla.org/show_bug.cgi?id=460801
-      setTimeout(() => {
+    let dropOperation = DROP_EFFECT_TO_DROP_OPERATION[state.dropEffect];
+    let items = readFromDataTransfer(e.dataTransfer);
+
+    let rect = (e.currentTarget as HTMLElement).getBoundingClientRect();
+    let event: DropEvent = {
+      type: 'drop',
+      x: e.clientX - rect.x,
+      y: e.clientY - rect.y,
+      items,
+      dropOperation
+    };
+    // Set drop effect in global DnD state for Chrome Android. TODO: file bug against chrome
+    // where onDragEnd always returns "none" as its drop effect.
+    setDropEffect(state.dropEffect);
+
+    // In the case where a drop happens on a non-collection drop target, track the element in which the drop was performed
+    setDroppedTarget(e.currentTarget as HTMLElement);
+
+    let dndStateSnapshot = {...getDnDState()};
+    // Wait a frame to dispatch the drop event so that we ensure the dragend event fires first.
+    // Otherwise, if onDrop removes the original dragged element from the DOM, dragend will never be fired.
+    // This behavior is consistent across browsers, but see this issue for details:
+    // https://bugzilla.mozilla.org/show_bug.cgi?id=460801
+    setTimeout(() => {
+      setDnDState(dndStateSnapshot);
+      if (typeof options.onDrop === 'function') {
         options.onDrop(event);
-      }, 0);
-    }
+      }
+      // TODO: for now don't clear global DnD state since useDrag's deferred onDragEnd needs the values
+      // set in options.onDrop (aka droppedTarget and droppedCollection). I'm a bit wary about a case where the
+      // user performs a drop from a non RSP collection into a RSP droppable collection and thus our onDragEnd won't fire
+      // and clear the global DnD state. Luckily it might not be that big of an issue if we clear on our drag start? Can't clear on
+      // document dragend listener since that occurs before our deffered onDrop/onDragEnd
```

Comment:
Open to opinions on how to handle this more elegantly. Something that clears the global state after a short delay if onDragEnd doesn't happen? A 2nd global state tracker (split global dnd state tracker into drag state and drop state, with each sharing information between each other but each only being used in their respective domains?)

Reply:
actually I think I can just check the snapshot for a dragged collection and if one doesn't exist then we know we can clear after drop